### PR TITLE
Move IAM

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,6 +16,7 @@ project("aos_core_common_cpp")
 option(WITH_TEST "build with test" OFF)
 option(WITH_COVERAGE "build with coverage" OFF)
 option(WITH_DOC "build with documenation" OFF)
+option(WITH_IAM_API "build core api iam lib" ON)
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply CMakeLists.txt:19]: Done

File: include/iamclient/types.hpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+#include <aos/common/tools/error.hpp>
+#include <aos/iam/certhandler.hpp>
+
+namespace aos::common::iamclient {
+
+/**
+ * MTLS credentials function.
+ */
+
+using MTLSCredentialsFunc = std::function<std::shared_ptr<grpc::ChannelCredentials>(
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply include/iamclient/types.hpp:29]: Done

File: include/iamclient/types.hpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+/**
+ * Configuration.
+ */
+struct Config {
+    std::string mIAMPublicServerURL;
+    std::string mCACert;
+};
+
+class CertSubscriberItf {
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply include/iamclient/types.hpp:40]: Done

File: .github/workflows/build_test.yaml [hunk 4]
```diff
@@ -41,8 +41,16 @@ jobs:
...
+          name: test-logs
+          path: build/Testing/Temporary/LastTest.log
+          if-no-files-found: ignore
+
       - name: Upload codecov report
         uses: codecov/codecov-action@v3
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
-          files: ./build/coverage.total
+          files: ./build/coverage.total
```
Comment (mlohvynenko) [inline .github/workflows/build_test.yaml:56]: please add empty line at the end

File: include/iamclient/publicservicehandler.hpp [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+     */
+    PublicServiceHandler() = default;
+
+    /**
+     * Initializes handler.
+     *
+     * @param cfg Configuration.
+     * @param certLoader Certificate loader.
+     * @param cryptoProvider Crypto provider.
+     * @param insecureConnection Insecure connection.
```
Comment (mlohvynenko) [inline include/iamclient/publicservicehandler.hpp:46]: `@return Error.`

File: include/iamclient/publicservicehandler.hpp [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        }
+
+        std::unique_ptr<grpc::ClientContext> mCtx {};
+        std::future<void>                    mFuture;
+        bool                                 mClose {};
+
+        std::unordered_set<iam::certhandler::CertReceiverItf*> mSubscribers;
+    };
+
+    constexpr static auto cIAMPublicServiceTimeout = std::chrono::seconds(10);
```
Comment (mlohvynenko) [inline include/iamclient/publicservicehandler.hpp:114]:
```suggestion
    static constexpr auto cIAMPublicServiceTimeout = std::chrono::seconds(10);
```

File: include/iamclient/types.hpp [hunk 7]
```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline include/iamclient/types.hpp:2]:
```suggestion
```

File: include/logger/logmodule.hpp [hunk 8]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline include/logger/logmodule.hpp:2]:
```suggestion
```

File: src/iamclient/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline src/iamclient/CMakeLists.txt:2]:
```suggestion
```

File: src/iamclient/publicservicehandler.cpp [hunk 10]
```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline src/iamclient/publicservicehandler.cpp:2]:
```suggestion
```

File: tests/iamclient/iamclient_test.cpp [hunk 11]
```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline tests/iamclient/iamclient_test.cpp:2]:
```suggestion
```

File: tests/iamclient/mocks/certsubscriber.hpp [hunk 12]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline tests/iamclient/mocks/certsubscriber.hpp:2]:
```suggestion
```

File: .github/workflows/build_test.yaml (hunk 4 above)
Comment (MykolaSolyanko) [reply .github/workflows/build_test.yaml:56]: Done

File: include/iamclient/types.hpp (hunk 7 above)
Comment (MykolaSolyanko) [reply include/iamclient/types.hpp:2]: Done

File: include/logger/logmodule.hpp (hunk 8 above)
Comment (MykolaSolyanko) [reply include/logger/logmodule.hpp:2]: Done

File: src/iamclient/CMakeLists.txt (hunk 9 above)
Comment (MykolaSolyanko) [reply src/iamclient/CMakeLists.txt:2]: Done

File: tests/iamclient/CMakeLists.txt [hunk 13]
```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (C) 2024 Renesas Electronics Corporation.
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply tests/iamclient/CMakeLists.txt:2]: Done

File: tests/iamclient/iamclient_test.cpp (hunk 11 above)
Comment (MykolaSolyanko) [reply tests/iamclient/iamclient_test.cpp:2]: Done

File: tests/iamclient/mocks/certsubscriber.hpp (hunk 12 above)
Comment (MykolaSolyanko) [reply tests/iamclient/mocks/certsubscriber.hpp:2]: Done

File: tests/iamclient/stubs/iamserver.hpp [hunk 14]
```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply tests/iamclient/stubs/iamserver.hpp:2]: Done

**Review by mlohvynenko — APPROVED** [review]

**Review by mykola-kobets-epam — APPROVED** [review]

File: include/iamclient/types.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (al1img) [reply include/iamclient/types.hpp]: I mean all types.hpp content?

[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply include/iamclient/types.hpp]: Done

**Pull request merged**